### PR TITLE
feat: ability to edit user settings in bulk

### LINKED ISSUE #492: Bulk User Permisson Edit Button
**Is your feature request related to a problem? Please describe.**
With the recent addition of separated auto approve permissions for Movies/Series. I have to manually edit everyone permission to update them, so they have auto approve movies option. This takes a lot of time and can be easily fixed with a bulk edit button

**Describe the solution you'd like**
Under the User tab, add a Bulk Edit Button. With this, the server admin can select everyone or a selected few and edit their permission.

**Describe alternatives you've considered**
Ombi User Management Page

**Additional context**

Screenshot_226

### PR DESCRIPTION
**Pull request by hmnd**
#### Description
Allow editing multiple users' settings at the same time in Users table.

#### Screenshot (if UI related)
35rf1Eii
tSfIds3l

#### Todos

- [x] Sucessfully builds `yarn build`
- [x] Translation Keys `yarn i18n:extract`

#### Issues Fixed or Closed by this PR

- Fixes #492

### CODE REVIEW

**Comment by hmnd** [comment]
Btw not sure whether it's alright to change key names in the localization files due to refactoring a component out, so let me know if that's wrong.

**Comment by sct** [comment]
> Btw not sure whether it's alright to change key names in the localization files due to refactoring a component out, so let me know if that's wrong.

It _should_ be fine but weblate tends to get have issues with merge conflicts when this happens. Maybe just modify only the `en.json` file and let the translators migrate the translations over.

**Comment by sct** [comment]
@hmnd This PR is really fantastic! I have a few comments I will leave in the review but I also wanted to point out a design choice that I don't necessarily agree with.

https://user-images.githubusercontent.com/234213/104017633-d1986a00-51fb-11eb-9217-6c4ae86320b8.mov

I really don't like how the whole table shifts here when clicking on this button. It can easily cause misclicks? And the effect is really jarring. What about moving the button to the right side of the table above the normal actions? You can even get away I think with using `buttonSize="sm"` there to make it fit pretty nicely!

Screen Shot 2021-01-08 at 21 54 10

What do you think?

File: src/components/UserList/index.tsx [hunk 1]
```diff
@@ -175,6 +256,20 @@ const UserList: React.FC = () => {
         <Table.TBody>
           {data?.map((user) => (
             <tr key={`user-list-${user.id}`}>
+              <Table.TD>
+                {user.id !== currentUser?.id && (
```
Comment (sct) [inline src/components/UserList/index.tsx:260]: What do you think about not allowing modification of user id 1 here either? (And in the logic to select all users)

**Comment by hmnd** [comment]
> I really don't like how the whole table shifts here when clicking on this button. It can easily cause misclicks? And the effect is really jarring. What about moving the button to the right side of the table above the normal actions? You can even get away I think with using `buttonSize="sm"` there to make it fit pretty nicely!

Totally agree. I couldn't think of a better place to put it that wouldn't move everything, but I think your idea is great!

**Comment by hmnd** [comment]
@sct Hmm how about one of these maybe? I don't really love how the button looks in the table header on the right. Imo it doesn't align well with the bigger buttons below it and isn't noticeable unless you scroll over to the side on screen widths < ~1400px.
image
image

**Comment by Magikarplvl4** [comment]
@hmnd Maybe "Bulk Edit" instead of only "edit"?

**Comment by sct** [comment]
Screen Shot 2021-01-09 at 19 03 16

I think it looks fine on the right? It's also where all the other edit buttons are. Yes, at lower resolutions that area gets cut off but the expected place for the edit buttons remains the same.

**Comment by sct** [comment]
I do like @Magikarplvl4's idea of having it say Bulk Edit:
Screen Shot 2021-01-09 at 20 11 48

**Comment by hmnd** [comment]
@sct Yeah I think that works well! The button disappearing and reappearing was what put me off, but it works much better disabled like that. Will make those changes soon. :)

**Comment by ankarhem** [comment]
Are you still working on this?

**Comment by hmnd** [comment]
@ankarhem yup, working on rebasing now :)

**Comment by hmnd** [comment]
@sct should be ready for your review again! Sorry for the wait.

**Review by sct — APPROVED** [review]

**Comment by rcohen-unext** [comment]
@all-contributors please add @hmnd for code

**Pull request merged**